core/src/main/java/hudson/model/Run.java

```diff
@@ -1509,11 +1509,11 @@ public String toString() {
     	}
     	
         String message = "No such file: " + logFile;
-    	return new ByteArrayInputStream(charset != null ? message.getBytes(charset) : message.getBytes());
+    	return new ByteArrayInputStream(charset != null ? message.getBytes(charset) : message.getBytes(Charset.defaultCharset()));
     }
    
     public @NonNull Reader getLogReader() throws IOException {
-        if (charset==null)  return new InputStreamReader(getLogInputStream());
+        if (charset==null)  return new InputStreamReader(getLogInputStream(),Charset.defaultCharset());
```

Comment:
I think this is OK. https://github.com/jenkinsci/jep/blob/master/jep/206/README.adoc means that for Pipeline this should always be UTF-8.